adr/2022_02_11_declaring_services_in_container.md

```diff
@@ -0,0 +1,65 @@
+# Declaring services in container
+
+* Status: accepted
+* Date: 2022-02-11
+
+## Context and Problem Statement
+
+Since Symfony 3.4, all Symfony services are declared private by default. And this is the recommended Symfony approach. On the 
+other hand, first stable Sylius release was published before that and therefore at the moment of publication of this recommendation
+Sylius had thousands (or at least hundreds) public services. The question is how should we handle services declared after the 
+3.4 release.
+
+## Decision Drivers
+
+Declared services should be:
+* services should be easily usable in ResourceControllers (where we are allowing for fetching services directly from container)
+* services should be easily usable in state machine callback (where we are allowing for fetching services directly from container)
+* it would be better to have one, coherent rule for all services
+* services should be easily accessible in PHPUnit
```

Comment:
Looking at these two points - maybe just remove `Declared services should be:` at the beginning, so we don't need any changes in the points